src/datasets/utils/_dill.py

```diff
@@ -162,11 +162,17 @@ def _save_torchTensor(pickler, obj):
     import torch  # type: ignore
 
     # `torch.from_numpy` is not picklable in `torch>=1.11.0`
-    def create_torchTensor(np_array):
-        return torch.from_numpy(np_array)
+    def create_torchTensor(np_array, dtype=None):
+        tensor = torch.from_numpy(np_array)
+        if dtype:
+            tensor = tensor.type(torch.bfloat16)
+        return tensor
```

Comment:
For future compatibility? :smile:

```suggestion
    def create_torchTensor(np_array, dtype=None):
        tensor = torch.from_numpy(np_array)
        if dtype:
            tensor = tensor.type(dtype)
        return tensor
```

Reply:
ah good catch, I went too fast x)

Reply:
fixed at https://github.com/huggingface/datasets/pull/7003